tests/test_inspect.py

```diff
@@ -31,72 +31,44 @@ def func2():
     def func3(x: Bar, y: Foo):
         pass
 
+    assert andi.inspect(Foo.__init__) == {}
     assert andi.inspect(func1) == {'x': [Foo]}
     assert andi.inspect(func2) == {}
     assert andi.inspect(func3) == {'x': [Bar], 'y': [Foo]}
 
-    assert andi.to_provide(func1, possible) == {'x': Foo}
-    assert andi.to_provide(func2, possible) == {}
-    assert andi.to_provide(func3, possible) == {'x': Bar, 'y': Foo}
-
-    # incomplete data
-    assert andi.to_provide(func1, {Baz}) == {}
-    assert andi.to_provide(func2, {Baz}) == {}
-    assert andi.to_provide(func2, set()) == {}
-    assert andi.to_provide(func3, {Baz}) == {}
-
 
 def test_union():
     def func(x: Union[Foo, Bar]):
         pass
 
     assert andi.inspect(func) == {'x': [Foo, Bar]}
 
-    assert andi.to_provide(func, {Foo, Bar}) == {'x': Foo}
-    assert andi.to_provide(func, {Bar}) == {'x': Bar}
-    assert andi.to_provide(func, {Baz}) == {}
-    assert andi.to_provide(func, set()) == {}
-
 
 def test_optional():
     def func(x: Optional[Foo]):
         pass
 
-    assert andi.inspect(func) == {'x': [Foo, None]}
-
-    assert andi.to_provide(func, {Foo, Bar}) == {'x': Foo}
-    assert andi.to_provide(func, {Foo, Bar, None}) == {'x': Foo}
-    assert andi.to_provide(func, {Bar}) == {}
-    assert andi.to_provide(func, {Bar, None}) == {'x': None}
+    assert andi.inspect(func) == {'x': [Foo, type(None)]}
 
 
 def test_optional_union():
     def func(x: Optional[Union[Foo, Baz]]):
         pass
 
-    assert andi.inspect(func) == {'x': [Foo, Baz, None]}
-
-    assert andi.to_provide(func, {Foo, Bar}) == {'x': Foo}
-    assert andi.to_provide(func, {Baz, Foo, None}) == {'x': Foo}
-    assert andi.to_provide(func, {Baz}) == {'x': Baz}
-    assert andi.to_provide(func, {Bar}) == {}
-    assert andi.to_provide(func, {Bar, None}) == {'x': None}
-    assert andi.to_provide(func, {}) == {}
-    assert andi.to_provide(func, {None}) == {'x': None}
```

Comment:
Where is plan generation for Union/Optional tested now?

Reply:
There were some tests in `test_plan.py` but I have refactored them added a few more in https://github.com/scrapinghub/andi/pull/8/commits/f2efaacde2743f7830c4f606f8411872ee83d341